pkg/restore/change_image_name_action.go

```diff
@@ -0,0 +1,275 @@
+			return nil, errors.Wrap(err, "error getting item's spec.containers")
+		}
+		for i, container := range containers {
+			a.logger.Infoln("container:", container)
+			if image, ok := container.(map[string]interface{})["image"]; ok {
+				imageName := image.(string)
+				if exists, newImageName, err := a.isImageReplaceRuleExist(log, imageName, config); exists && err == nil {
+					needUpdateObj = true
+					a.logger.Infof("Updating item's image from %s to %s", imageName, newImageName)
+					container.(map[string]interface{})["image"] = newImageName
+					containers[i] = container
+				}
+			}
+		}
+
+		if needUpdateObj {
+			if err := unstructured.SetNestedField(obj.UnstructuredContent(), containers, "spec", "template", "spec", "containers"); err != nil {
+				return nil, errors.Wrap(err, "unable to set item's container image")
+			}
+			a.logger.Infof("obj.Object %v", obj.Object)
+		}
+
+		//handle initContainers
+		needUpdateObj = false
+		initContainers, _, err := unstructured.NestedSlice(obj.UnstructuredContent(), "spec", "template", "spec", "initContainers")
+		if err != nil {
+			return nil, errors.Wrap(err, "error getting item's spec.initContainers")
+		}
+		for i, container := range initContainers {
+			if image, ok := container.(map[string]interface{})["image"]; ok {
+				imageName := image.(string)
+				if exists, newImageName, err := a.isImageReplaceRuleExist(log, imageName, config); exists && err == nil {
+					needUpdateObj = true
+					a.logger.Infof("Updating item's image from %s to %s", imageName, newImageName)
+					container.(map[string]interface{})["image"] = newImageName
+					initContainers[i] = container
+				}
+			}
+		}
+
+		if needUpdateObj {
+			if err := unstructured.SetNestedField(obj.UnstructuredContent(), initContainers, "spec", "template", "spec", "initContainers"); err != nil {
+				return nil, errors.Wrap(err, "unable to set item's initContainer image")
+			}
+		}
+
+	}
+	return velero.NewRestoreItemActionExecuteOutput(obj), nil
+}
+
+func (a *ChangeImageNameAction) isImageReplaceRuleExist(log *logrus.Entry, oldImageName string, cm *corev1.ConfigMap) (exists bool, newImageName string, err error) {
+	if oldImageName == "" {
+		log.Infoln("Item has no old image name specified")
+		return false, "", nil
+	}
+	log.Debug("oldImageName: ", oldImageName)
+
+	//if image name have letters that configmap not support for key
+	//use "image-delimiter": <delimiter> the value can be any words that can
+	//be used as delimiter
+	//for current implementation the <delimiter> value can only be ","
+	//use "case1": "<old_image_name_sub_part><delimiter><new_image_name_sub_part>"
+	//e.x: in case your old image name is 1.1.1.1:5000/abc:test
+	//     "image-delimiter": ","
+	//     "case1":"1.1.1.1:5000,2.2.2.2:3000"
+	//     "case2":"5000,3000"
+	//     "case3":"abc:test,edf:test"
+	//     "case4":"1.1.1.1:5000/abc:test,2.2.2.2:3000/edf:test"
+	if delimiter, ok := cm.Data[DELIMITER_KEY]; !ok {
```

Comment:
Now that we're only using "," as the delimiter, I think this section is unnecessary. This code errors out if DELIMITER_KEY is unspecified, and then errors out if it's specified to anything other than ",". I think it would be easier if we just removed it entirely and just reference DELIMITER_VALUE rather than delimiter below.

Reply:
ok. 
DELIMITER_KEY  related code is deleted. new code test passed.
Thanks for you comments.